src/System.Diagnostics.TraceSource/src/System/Diagnostics/DefaultTraceListener.cs

```diff
@@ -160,13 +151,23 @@ private void WriteLine(string message, bool useLogFile)
 
         private void Write(string message, bool useLogFile)
         {
-            if (NeedIndent && !string.IsNullOrEmpty(message)) 
+            if (message == null)
```

Comment:
As seen in Desktop and existing Debug.Write, Trace.Write("") should skip indentation and just write empty string